routes/register.js

```diff
@@ -0,0 +1,34 @@
+const express = require('express');
+const router = express.Router();
+const User = require('../models/user')
+const bcrypt = require('Bcryptjs')
+const jwt = require('jsonwebtoken')
+
+router.post('/register', async (req, res) => {
+    const {
+        username,
+        password,
+        email,
+        lastName,
+        firstName
+        } = req.body
+
+    let user = await User.findOne({email})
+
+    if (user) {
+        return res.redirect('/register');
+    }
+ 
+    const hashedPass = await bcrypt.hash(password, 12);
+
+    user = new User({
+        username,
+        password: hashedPass,
+        email,
+        lastName,
+        firstName
+    })
+    await user.save();
+    res.redirect('/')
+});
+    module.exports = router;
```

Comment:
Too many duplicates, you already have 2 register routes and another as a controller, what on earth is going on lol

Reply:
its the one that will work on publicJS